File: OCDemo/OCDemo/设计模式/Cat.m
//
//  Cat.m
//  OCDemo
//
//  Created by mkrq-yh on 2019/3/21.
//  Copyright © 2019 mkrq-yh. All rights reserved.
//

#import "Cat.h"

@implementation Cat

- (void)printClassName {
    NSLog(@"%@---%@",NSStringFromClass([self class]),self);
}

- (void)catchTheMouse {
    NSLog(@"抓老鼠");
}

@end

File: SwiftDemo/SwiftDemo/设计模式/Cat.swift
//
//  Cat.swift
//  SwiftDemo
//
//  Created by mkrq-yh on 2019/3/21.
//  Copyright © 2019 mkrq-yh. All rights reserved.
//

import UIKit

class Cat: Animal {

    func catchTheMouse() {
        print("抓老鼠")
    }
    
    override func printClassName() {
        print("\(NSStringFromClass(self.classForCoder))")
    }
}
